src/mds/Server.cc

```diff
@@ -1085,8 +1085,9 @@ void Server::recover_filelocks(CInode *in, bufferlist locks, int64_t client)
  */
 void Server::recall_client_state(float ratio)
 {
-  int max_caps_per_client = (int)(g_conf->mds_cache_size * .8);
-  int min_caps_per_client = 100;
+  int max_caps_per_client = (int)(g_conf->mds_cache_size * g_conf->max_ratio_caps_per_client);
+  int min_caps_per_client = (int)(g_conf->min_caps_per_client);
+  assert(min_caps_per_client < max_caps_per_client);
```

Comment:
Agree, I add the derr hint, and adjust max_caps_per_client.
thanks for you reviews.